themes/default/content/blog/pulumi-kubernetes-new-2022/index.md

```diff
@@ -0,0 +1,281 @@
+3. Support for platforms like Azure DevOps that require specific Git implementations
+4. Sources and Stacks can be defined by different teams, since they are separate resources which can have independent RBAC rules applied.
+5. Many advanced Git features including: following semver ranges, excluding files from source, recursively cloning submodules, verifying revision signatures, and more.
+
+To use the Pulumi Kubernetes Operator with Flux, install each of the necessary controllers for Flux and the Pulumi Kubernetes Operator, and then the necessary `Role` and `RoleBindings` to allow the Pulumi Kubernetes Operator to read from Flux sources.
+
+```yaml
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: flux-source-access
+rules:
+- apiGroups:
+  - source.toolkit.fluxcd.io
+  resources:
+  - "*"
+  verbs:
+  - get
+  - list
+  - watch
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: pulumi-flux-source-access
+subjects:
+- kind: ServiceAccount
+  name: pulumi-kubernetes-operator
+roleRef:
+  kind: Role
+  name: flux-source-access
+  apiGroup: rbac.authorization.k8s.io
+```
+
+Once those pre-requisites have been met, simply deploy the Flux Source (in this case a `GitRepository`) and Pulumi `Stack` resource that define your infrastructure delivery:
+
+```yaml
+---
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: GitRepository
+metadata:
+  name: pkodev
+spec:
+  interval: 30s
+  ref:
+    branch: main
+  url: https://github.com/lukehoban/pko-dev
+---
+apiVersion: pulumi.com/v1
+kind: Stack
+metadata:
+  name: pkodev
+spec:
+  stack: lukehoban/pkodev/test
+  fluxSource:
+    sourceRef:
+      apiVersion: source.toolkit.fluxcd.io/v1beta2
+      kind: GitRepository
+      name: pkodev
+    dir: basic/
+  destroyOnFinalize: true
+```
+
+Note that by using the new `fluxSource` property on a `pulumi.com/v1/Stack` resource, you can configure it to use the corresponding source instead of an inline definition of the target Git repo and branch.  The `GitRepository` could be replaced with any other Flux Source.  For example, to pull our source code from an S3 bucket, use this instead:
+
+```yaml
+---
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: Bucket
+metadata:
+  name: pkodev
+spec:
+  interval: 5m0s
+  provider: aws
+  bucketName: podinfo
+  endpoint: s3.amazonaws.com
+  region: us-east-1
+  timeout: 30s
+---
+apiVersion: pulumi.com/v1
+kind: Stack
+metadata:
+  name: pkodev
+spec:
+  stack: lukehoban/pkodev/test
+  fluxSource:
+    sourceRef:
+      apiVersion: source.toolkit.fluxcd.io/v1beta2
+      kind: Bucket
+      name: pkodev
+    dir: basic/
+  destroyOnFinalize: true
+```
+
+If we are using the `GitRepository` source along with GitHub, we can use Flux features like [WebHook Recievers] to expose a webhook that GitHub can trigger on pushes so that updates will be pushed into Flux and the Pulumi Kubernetes Operator directly (insead of waiting to pull them):
```

Comment:
```suggestion
If we are using the `GitRepository` source along with GitHub, we can use Flux features like [WebHook Recievers](https://fluxcd.io/flux/guides/webhook-receivers/) to expose a webhook that GitHub can trigger on pushes so that updates will be pushed into Flux and the Pulumi Kubernetes Operator directly (instead of waiting to pull them):
```